### PR TITLE
Avoid exception when setting the network stream timeout

### LINKED ISSUE #17781: [BUG] Perf regression in Azure.Core 1.0.2 -> 1.1.0
>1.	Azure.Core 1.0.2 -> 1.1.0 (6548 -> 5693 ops/sec)

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/17781

Also avoids an annoying exception that shows up in VS Debug Output windows constantly.

Before (using `dotnet run -c Release -f netcoreapp2.1 -- Download -s 10000`):
```

=== Test ===
Current         Total
3206            3206
3354            6560
3291            9851
3323            13174
3371            16545
3434            19979
3563            23542
3677            27219
3323            30542
3265            33807

=== Results ===
Completed 33807 operations in a weighted-average of 10.01s (3,378.15 ops/s, 0.000 s/op)
```

After:
```
=== Test ===
Current         Total
4182            4182
4169            8351
4153            12504
4268            16772
4246            21018
4158            25176
4195            29371
4199            33570
4270            37840
3851            41691

=== Results ===
Completed 41691 operations in a weighted-average of 10.02s (4,160.27 ops/s, 0.000 s/op)
```

**Azure.Core 1.0.2**

```
=== Test ===
Current         Total
4215            4215
4265            8480
4105            12585
4679            17264
4376            21640
4385            26025
4132            30157
4518            34675
4575            39250
3728            42978

=== Results ===
Completed 42978 operations in a weighted-average of 10.00s (4,296.94 ops/s, 0.000 s/op)
```

### CODE REVIEW

File: sdk/core/Azure.Core/perf/Azure.Core.Perf.csproj [hunk 1]
```diff
@@ -5,9 +5,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="../../Azure.Core/src/Azure.Core.csproj" />
+    <ProjectReference Condition="'$(AzureCoreVersion)' == ''" Include="../../Azure.Core/src/Azure.Core.csproj" />
```
Comment (pakrym) [inline sdk/core/Azure.Core/perf/Azure.Core.Perf.csproj:8]: Allows easy version switching

File: sdk/core/Azure.Core/perf/Azure.Core.Perf.csproj [hunk 2]
```diff
@@ -5,9 +5,11 @@
...
+    <PackageReference Condition="'$(AzureCoreVersion)' != ''" Include="Azure.Core" VersionOverride="$(AzureCoreVersion)" />
+  </ItemGroup>
+  <ItemGroup>
     <ProjectReference Include="$(MSBuildThisFileDirectory)..\..\..\..\common\Perf\Azure.Test.Perf\Azure.Test.Perf.csproj" />
-    <ProjectReference Include="$(AzureCoreTestFramework)" />
```
Comment (pakrym) [inline sdk/core/Azure.Core/perf/Azure.Core.Perf.csproj:10]: Need to drop extra dependencies that prevent azure.core from being downgraded

**Review by kasobol-msft — APPROVED** [review]

File: sdk/core/Azure.Core/src/Pipeline/Internal/ReadTimeoutStream.cs [hunk 3]
```diff
@@ -138,7 +138,10 @@ private void UpdateReadTimeout()
         {
             try
```
Comment (mikeharder) [inline sdk/core/Azure.Core/src/Pipeline/Internal/ReadTimeoutStream.cs:139]: Do you still need the try/catch if you are checking `CanTimeout`?

Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/Internal/ReadTimeoutStream.cs:139]: I wanted to keep it for some possible badly-behaved streams.

**Pull request merged**